### ISSUE TITLE
Hebrew Translation Progress

### ISSUE THREAD

**Issue by tesseralis** [reactjs/he.reactjs.org#1]
## Maintainer List

* @galnir
* @galtalmor 
* @itayyehezkel 

## For New Translators

To translate a page:

1. Check that no one else has claimed your page in the checklist and comments below.
2. Comment below with the name of the page you would like to translate. **Please take only one page at a time**.
3. Clone this repo, translate your page, and submit a pull request!

Before contributing, read the glossary and style guide (once they exist) to understand how to translate various technical and React-specific terms.

Please be prompt with your translations! If you find find that you can't commit any more, let the maintainers know so they can assign the page to someone else.

## For Maintainers

When someone volunteers, edit this issue with the username of the volunteer, and with the PR. Ex:

* [x] Home Page (@tesseralis) #1

When PRs are merged, make sure to mark that page as completed!

## Core Pages

To do before releasing as an "official" translation. **Please translate these pages first.**

* [x] Home Page (@gmadar) #4 
* [x] Tutorial (@galtalmor) #10 

### Basics

* [x] Hello World (@bromy) #8
* [x] Introducing JSX (@galtalmor) #42
* [x] Rendering Elements (@dormd) #21
* [x] Components and Props (@galtalmor) #20
* [x] State and Lifecycle (@galtalmor) #25
* [x] Handling Events (@galtalmor) #26
* [x] Conditional Rendering (@galtalmor) #30
* [x] Lists and Keys (@itayyehezkel) #36 
* [x] Forms (@galtalmor) #32
* [x] Lifting State Up (@galtalmor) #34
* [x] Composition vs Inheritance (@dormd) #40
* [x] Thinking in React (@galnir) #14

### API Reference

* [ ] React
* [x] React.Component (@itzikya) #65 
* [x] ReactDOM (@galtalmor) #45
* [x] ReactDOMServer (@galtalmor) #43
* [x] DOM Elements (@galnir) #46
* [x] SyntheticEvent (@galnir) #51 
* [x] Test Utilities (@galnir) #52 
* [x] Shallow Renderer (@itayyehezkel) #39 
* [x] Test Renderer (@galtalmor) #93
* [x] JS Environment Requirements (@itayyehezkel) #37 
* [x] Glossary (@galtalmor) #49 

### Navigation

These are the navigation links that appears in the sidebar. Possibly wait until
the corresponding sections are translated to do these.

* [x] tutorial/nav.yml (@galtalmor) #10 
* [x] docs/nav.yml (@galtalmor) #13

## Next Steps

These are the next most important translations, ordered by importance:

### Installation

* [x] Getting Started (@galtalmor) #15
* [x] Add React to a Website (@galnir) #51  
* [x] Create a New React App (@galnir) #55 
* [x] CDN Links (@dormd) #31

### Hooks

Everyone is hooked on hooks! People are really excited for these and it would be great for them to be translated.

  * [x] Introducing Hooks (@itayyehezkel) #18 (@galnir) #265
  * [x] Hooks at a Glance (@galnir ) #94 
  * [x] Using the State Hook (@galnir) #57 
  * [x] Using the Effect Hook (@galnir) #73 
  * [x] Rules of Hooks (@galnir) #84 
  * [x] Building Your Own Hooks (@galnir) #101 
  * [x] Hooks API Reference (@galnir) #113 
  * [ ] Hooks FAQ (@galnir)

### Warnings

These are the pages that you get when you click the links in the console (e.g. https://reactjs.org/warnings/dont-call-proptypes.html). People tend to search these, so **please don't translate the error message itself**.
 
* [x] dont-call-proptypes (@galnir)
* [x] invalid-aria-prop (@itayyehezkel) #112 
* [ ] invalid-hook-call-warning
* [ ] legacy-factories
* [ ] refs-must-have-owner
* [x] special-props (@itayyehezkel) #111 
* [ ] unknown-prop

### Advanced Guides

* [x] Accessibility (@rgrnwd) #71 
* [x] Code-Splitting (@rgrnwd) #92 
* [x] Context (@rgrnwd) #77 
* [x] Error Boundaries (@rgrnwd) #74 
* [x] Forwarding Refs (@rgrnwd) #77 
* [x] Fragments (@itayyehezkel) #109 
* [x] Higher-Order Components (@rgrnwd) #95 
* [ ] Integrating with Other Libraries
* [ ] JSX In Depth
* [ ] Optimizing Performance
* [ ] Portals
* [ ] React Without ES6
* [ ] React Without JSX (@tsehori)
* [ ] Reconciliation
* [ ] Refs and the DOM
* [ ] Render Props
* [ ] Static Type Checking
* [ ] Strict Mode
* [ ] Typechecking With PropTypes
* [x] Uncontrolled Components (@itayyehezkel) #110 
* [ ] Web Components

### FAQ

* [x] APIs and AJAX (@itayyehezkel) #62 
* [x] Babel, JSX, and Build Steps (@itayyehezkel) #63 
* [x] Passing Functions to Components (@orrsteinberg) #172
* [ ] Component State
* [x] Styling and CSS (@itayyehezkel) #64 
* [x] File Structure (@galnir) #44
* [ ] Versioning Policy
* [ ] Virtual DOM and Internals

### Contributing

Priority: Low

* [ ] How to Contribute
* [ ] Codebase Overview
* [ ] Implementation Notes
* [ ] Design Principles

### Components

Components in `src/components` that have some text in them.

* [x] CodeEditor (@itayyehezkel) #27 
* [x] CodeExample (@itayyehezkel) #28 
* [x] ErrorDecoder (@galnir) #23 
* [x] LayoutFooter (@galtalmor) #13
* [x] LayoutHeader (@galtalmor) #11
* [x] MarkdownPage (@itayyehezkel) #19 

## Additional Translations

These are not the primary translation targets.

* [ ] Blog
* [ ] Community

**Comment by gmadar** [comment]
* Home Page

**Comment by bentlusty** [comment]
@gmadar Added you to Home page. thank you!

**Comment by galtalmor** [comment]
* Tutorial

**Comment by galnir** [comment]
@gmadar Added you to the tutorial

**Comment by galtalmor** [comment]
> @gmadar Added you to the tutorial

Wrong mention, bug tagging is okay :)

**Comment by galnir** [comment]
@galtalmor oops sorry

**Comment by bromy** [comment]
- [ ] Hello World

**Comment by galnir** [comment]
@bromy It's yours 👍

**Comment by galnir** [comment]
Does anyone have a solution for fixing punctuation when translating into hebrew? There is an example in gmadar's pull request

**Comment by galtalmor** [comment]
> Does anyone have a solution for fixing punctuation when translating into Hebrew?
I suppose it will just require some CSS tweaking with `direction: rtl`, but I'm not sure how would that works with the way they build and publish this website. I think it is best to try and contact someone from the team in charge to figure out if it is okay to make changes in the website's CSS since this problem will exist everywhere with this translation.

**Comment by tesseralis** [comment]
@galtalmor It's definitely okay to make changes to the CSS -- Arabic and Persian dealt with the same thing and actually have PRs to do it: https://github.com/reactjs/ar.reactjs.org/pull/8

You might want to post there to see if they can help you set it up for Hebrew!

**Comment by galnir** [comment]
@galtalmor A member of the Arabic translation team already helped us with that :) https://github.com/reactjs/he.reactjs.org/pull/6

**Comment by galtalmor** [comment]
> @galtalmor A member of the Arabic translation team already helped us with that :) #6

👍

**Comment by dormd** [comment]
Awesome idea 😍
Rendering Elements

**Comment by galnir** [comment]
@dormd It's yours :)

**Comment by galtalmor** [comment]
* Getting Started

**Comment by itayyehezkel** [comment]
Introducing Hooks

**Comment by galtalmor** [comment]
Hey @bentlusty & @galnir I think you guys should update this main thread, it's starting to be hard to follow on what is already completed, what's not and who's working on stuff. I wouldn't mind helping with that if you guys are too busy. Thanks!

**Comment by galtalmor** [comment]
Here's the list of completed tasks with related PRs:
* [x]  Home Page (@gmadar) #4 
* [x]  Hello World (@bromy) #8 
* [x]  Tutorial (@galtalmor) #10 
* [x]  tutorial/nav.yml (@galtalmor) #10 
* [x]  LayoutHeader (@galtalmor) #11 
* [x]  LayoutFooter (@galtalmor) #13 
* [x] Thinking in React (@galnir) #14 
* [x] Getting Started (@galtalmor) #15

**Comment by galtalmor** [comment]
* Components and Props

**Comment by galnir** [comment]
@tesseralis Can you please add @galtalmor as a maintainer? It would really help us keep up with the work

**Comment by tesseralis** [comment]
@galnir sure! In the future you can [open a PR](https://github.com/reactjs/reactjs.org-translation#adding-a-maintainer) if you need to add more maintainers.

**Comment by tesseralis** [comment]
@galtalmor once you accept the invite to join the reactjs org you should be able to update this issue :)

**Comment by galtalmor** [comment]
> @galtalmor once you accept the invite to join the reactjs org you should be able to update this issue :)

@tesseralis Yep, it works. Thanks!

**Comment by bromy** [comment]
As you are all working on these, I was wondering about what setup or process you are using in your text editor. Working with the mixed left-to-right/right-to-left text files made the formatting more of a challenge than it should have been 😄. Text kept jumping around, highlighting/copying/pasting felt unpredictable, and the markdown syntax was easily thrown off. Are there any editor or OS settings that made it more intuitive to work with the bidirectional nature of the text?

**Comment by galnir** [comment]
@bromy My main text editor is vscode, but it doesn't support right to left(in my experience). so for translation I use notepad++

**Comment by galtalmor** [comment]
@bromy I use WebStorm, it doesn't really support RTL (doesn't let you control it or stick the code to the right), but when the first word is in hebrew it kinda justifies the content in RTL, so it is good enough for me. Also it got a markup plugin that lets you see changes on the fly, so that is also pretty helpful.
I don't think there will be any really good solution here, but I will be happy to be surprised.

**Comment by dormd** [comment]
CDN Links

**Comment by itayyehezkel** [comment]
MarkdownPage

**Comment by dormd** [comment]
Composition vs Inheritance

**Comment by bentlusty** [comment]
Hey guys, I'm sorry I wasn't here as much as I should. Regrettably I don't have the time to maintain this repo. 

See ya on rothschild boulevard :)

**Comment by dormd** [comment]
Hooks at a Glance

**Comment by galtalmor** [comment]
@tesseralis How often do languages websites get published? On the English website there's a link to the Hebrew version, but the Hebrew version was never published, the live version is all in English. Is there a way for us to publish a new version?

**Comment by tesseralis** [comment]
That's actually something that's confusing me too. The other languages are working properly -- I think it's just broken. Are you all publishing off the `master` branch? @bvaughn do you think you could look into this?

In the meanwhile, if you want something to work off, you could just create another netlify app and set it to the repo

**Comment by bvaughn** [comment]
Hm...looks like the "production branch" was configured incorrect in Circle for some reason. I've changed it to "master" and triggered a manual deployment since it's pretty stale. Should be okay now?

**Comment by galtalmor** [comment]
@tesseralis @bvaughn Yes, now it seems to be working. Thanks!

**Comment by tesseralis** [comment]
We're almost done everyone!

@itayyehezkel @galtalmor what's the status of your pages?

Also, could anyone take the React.Component under "API"? It's the last of the core pages we need to do to mark this translation as complete!

**Comment by itzikya** [comment]
React.Component

**Comment by galnir** [comment]
@itzikya Added you!

**Comment by galtalmor** [comment]
Hey @tesseralis , I started a new job and had a few crazy weeks. I will complete the translation of my page soon and also review @galnir 's work. Cheers!

**Comment by rgrnwd** [comment]
Hey, I'm keen to contribute. I'll start with the Accessibility page, if that's ok with everyone? :)

**Comment by galnir** [comment]
@rgrnwd That's great, thanks!

**Comment by rgrnwd** [comment]
Can I pick up Error Boundaries next? Thanks!

**Comment by galnir** [comment]
@rgrnwd 👍

**Comment by rgrnwd** [comment]
Context & Forwarding Refs next 😃

**Comment by galnir** [comment]
@tesseralis hey, should we also translate pages like 404/versions/acknowlegements?

**Comment by tsehori** [comment]
Hey, can I pick up  Uncontrolled Components and React Without JSX?
Thanks!

**Comment by galnir** [comment]
@tsehori Ofc 👍

**Comment by tesseralis** [comment]
@itayyehezkel @galtalmor I'm going to open up your pages `React` and `TestRenderer` for now since it's been a while, so if anyone wants to take those pages to finish up the core translation, now's your chance!

@galnir I'm not sure which page that is? Do you have a link to the english react website for it?

**Comment by galnir** [comment]
@tesseralis
404 (page not found): https://reactjs.org/dhxnxdj

**Comment by tesseralis** [comment]
Yeah! That should be translated.

**Comment by rgrnwd** [comment]
Can I translate Code-Splitting and higher order components?

**Comment by tesseralis** [comment]
@galtalmor how is the API->React translation going? It's the last one we need to mark this as completed so we're eager to get that done!

**Comment by galtalmor** [comment]
@tesseralis Done about half of it. Will get it done this week.

**Comment by orrsteinberg** [comment]
Hey, I'd be happy to contribute. Is there anything that has high priority/that would be particularly helpful?
If not, I'll start with Passing Functions to Components in the FAQ section if that's alright.

**Comment by galtalmor** [comment]
@orrsteinberg anything that isn't complete yet

**Comment by orrsteinberg** [comment]
Got it 👍 I'll get right on it then.

**Comment by owolfy** [comment]
I saw some typos and inconsistent translations.
What is the convention for revisions PR?

**Comment by galnir** [comment]
> I saw some typos and inconsistent translations.
> What is the convention for revisions PR?

Open a PR with the fixes, nothing special

**Comment by galnir** [comment]
#280

**Comment by gaearon** [comment]
Thanks to everyone who worked on the old translation!
I've updated the TODO list for the new one in https://github.com/reactjs/he.react.dev/issues/280.
